src/common/utils/resolve_reference.py

```diff
@@ -103,16 +100,23 @@ def get_entry_point(self, data_source: DataSource) -> Tuple[dict, str]:
             )
         return result[0], result[0]["_id"]
 
-    def get_child(self, document: dict | list, data_source: DataSource, get_data_source: Callable) -> Tuple[Any, str]:
+    def get_child(
+        self, document: dict | list, data_source: DataSource, get_data_source: Callable, resolve: bool = False
+    ) -> Tuple[Any, str]:
         # Search inside an existing document (need to resolve any references first before trying to compare against filter)
         elements = [
-            resolve_reference(f"/{data_source.name}/{f['address']}", get_data_source).entity if is_reference(f) else f
+            resolve_reference(f"/{data_source.name}/{f['address']}", get_data_source, resolve).entity
```

Comment:
We should always resolve the references before searching in them, I think. Or can you think of a reason why we'd need queries like `[(type=reference)]`?